osx_utils.sh

```diff
@@ -475,12 +491,23 @@ function fuse_macos_intel_arm64 {
 
 function wrap_wheel_builder {
     if [[ "${PLAT:-}" == "universal2" ]]; then
-        (macos_intel_build_setup && $@)
-        rm -rf *-stamp
-        (macos_arm64_build_setup && $@)
-        fuse_macos_intel_arm64
+        if [[ "$(uname -m)" == "arm64" ]]; then
+            (macos_intel_cross_build_setup && $@)
+            rm -rf *-stamp
+            (macos_arm64_native_build_setup && $@)
+            fuse_macos_intel_arm64
+        else
+            (macos_intel_native_build_setup && $@)
+            rm -rf *-stamp
+            (macos_arm64_cross_build_setup && $@)
+            fuse_macos_intel_arm64
+        fi
```

Comment:
```suggestion
        else
            (macos_intel_native_build_setup && $@)
            rm -rf *-stamp
            (macos_arm64_cross_build_setup && $@)
        fi
        fuse_macos_intel_arm64
```
Moving the duplicate line outside of the block